tests/Simplify/EqualTest.elm

```diff
@@ -934,6 +934,787 @@ a = ({ a = 1 }).a == ({ a = 1 }).b
+a : Bool
+a = (l |> (not << List.isEmpty))
+"""
+                        ]
+        , test "should replace 0 /= (l |> List.length) with (not << List.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 /= (l |> List.length)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << List.isEmpty)"
+                            , details =
+                                [ "Whereas List.length takes as long to run as the number of elements in the List,"
+                                , "List.isEmpty runs in constant time."
+                                ]
+                            , under = "0 /= (l |> List.length)"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (l |> (not << List.isEmpty))
+"""
+                        ]
+        ]
+
+
+setTests : Test
+setTests =
+    describe "Set.size should be Set.isEmpty"
+        [ test "should replace Set.size s == 0 with Set.isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = Set.size s == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to Set.isEmpty"
+                            , details =
+                                [ "Whereas Set.size takes as long to run as the number of elements in the Set,"
+                                , "Set.isEmpty runs in constant time."
+                                ]
+                            , under = "Set.size s == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = Set.isEmpty s
+"""
+                        ]
+        , test "should replace size s == 0 with isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+import Set as CoreSet exposing (..)
+
+
+a : Bool
+a = size s == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to isEmpty"
+                            , details =
+                                [ "Whereas size takes as long to run as the number of elements in the Set,"
+                                , "isEmpty runs in constant time."
+                                ]
+                            , under = "size s == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Set as CoreSet exposing (..)
+
+
+a : Bool
+a = isEmpty s
```

Comment:
I think you wanted to have a test where you'd end up having to insert `CoreSet.isEmpty`, but you ended up not having one.

Maybe add a test with this code:
```elm
import Set as CoreSet exposing (size)

a : Bool
a = size s == 0
```

Reply:
Thought of this over breakfast this morning actually.

Reply:
I spend a lot of time testing your helper functions rather than the meat of my replacement, so that's part of the whole test pruning thing I think I want to do.

Reply:
Feel free to add comments to the helper functions if you think they're unclear!
I'm not sure whether you need to do pruning, it's okay to have multiple similar tests :man_shrugging:

Reply:
Sorry, I meant that the bulk of the runtime of these tests spends time on nuances of the helper functions. No, they were incredibly helpful and clear!!